### DISCUSSION TITLE
How to handle data change event

### DISCUSSION THREAD

**Discussion by WBangash** [gijzelaerr/python-snap7#397] [Q&A] [answered]
Hi Everyone,

I have used s7 library in C# where you define S7DataItem and  listen to DataChange event.
Can any one please help me how to do it in python.?
Thanks

**Comment by swamper123** [comment, marked as answer]
There is no out-of-the-box solution for your question and it still would take some time to implement it.

You may want to look for "Observer Patterns" for your purpose and make your own "better S7DataItem" class.

**Comment by gijzelaerr** [comment]
Closing as answered. There is no built-in data change event mechanism. Users need to implement their own polling/observer pattern.
